packages/graphql/src/schema/resolvers/defaultField.ts

```diff
@@ -25,11 +25,13 @@ import { GraphQLResolveInfo } from "graphql";
  */
 // eslint-disable-next-line consistent-return
 export default function defaultFieldResolver(source: any, args: any, context: unknown, info: GraphQLResolveInfo) {
-    const responseKey = info.fieldNodes[0].alias ? info.fieldNodes[0].alias.value : info.fieldNodes[0].name.value;
     if ((typeof source === "object" && source !== null) || typeof source === "function") {
-        const property = source[responseKey];
+        const property =
+            info.fieldNodes[0].alias && Object.prototype.hasOwnProperty.call(source, info.fieldNodes[0].alias.value)
```

Comment:
If there is `info.fieldNodes[0].alias` wouldnt there always be `info.fieldNodes[0].alias.value` ? I ask because this turnay operator is so scary too look at 😱 and could we just replace with: 

```js
info.fieldNodes[0].alias && source[info.fieldNodes[0].alias.value] : source[info.fieldNodes[0].name.value]
```

Reply:
Reading your comment, I would have thought so, it even matches the TypeScript definition!

But the tests fail when I switch it over, so perhaps not. 😞